src/handlers/books.rs

```diff
@@ -9,7 +9,18 @@ use super::Handler;
 
 impl Handler {
     pub async fn create_book(&self, book: CreateBook) -> Result<Book> {
-        self.book_repository.create_book(book).await
+        let created_book = self.book_repository.create_book(book.clone()).await;
+
+        if let Some(collection) = book.collection_id {
+            let collection_validated = self.get_collection_by_id(collection).await?;
+
+            let _ = &self.collection_repository.insert_book_into_collection(
```

Comment:
Isso é realmente necessário? se não houver atribuição a variavel o valor é dropado automaticamente.
```suggestion
            self.collection_repository.insert_book_into_collection(
```

Reply:
Inicialmente eu tinha ido nesse caminho que vc sugeriu, porém se eu não utilizar a "let _" o compilador está jogando um warning:
unused borrow that must be used
`#[warn(unused_must_use)]` on by defaultrustc
books.rs(17, 13): use `let _ = ...` to ignore the resulting value: `let _ = `